javers-spring/src/main/java/org/javers/spring/auditable/aspect/JaversAuditableRepositoryAspect.java

```diff
@@ -93,4 +120,55 @@ private void applyVersionChange(RepositoryMetadata metadata, Object domainObject
         handler.handle(metadata, domainObject);
     }
 
+    /**
+     * Returns TRUE if the {@link Throwable} matches against the current {@link JoinPoint}s method with Spring's @{@link Transactional}
+     * noRollbackFor or noRoolbackForClassName values.
+     *
+     * @param ex the {@link Throwable} which was thrown
+     * @param pjp the {@link JoinPoint} we are currently at
+     * @return TRUE if the advised method has @{@link Transactional} and will commit despite this @{@link Throwable}
+     */
+    protected boolean isStillExecute(JoinPoint pjp, Throwable ex) {
+        //Grab the current method signature we are operating on, and the Spring Transactional Annotation, if any
+        MethodSignature signature = (MethodSignature)pjp.getSignature();
+        Transactional transactionalAnnotation = signature.getMethod().getAnnotation(Transactional.class);
+
+        // If there is no Spring @Transactional annotation on the method, then we don't want to execute the normal
+        // versioning code
+        if( transactionalAnnotation == null ) {
+            return false;
+        }
+
+        boolean executeCommit = false;
+        Class<? extends Throwable> exceptionClass = ex.getClass();
+        //grab the @Transactional noRollbackFor classes
+        Class<? extends Throwable>[] noRollbackForClasses = transactionalAnnotation.noRollbackFor();
+        if( noRollbackForClasses != null && noRollbackForClasses.length > 0 ) {
+            for( Class<? extends Throwable> aClass : noRollbackForClasses ) {
```

Comment:
This looks quite complex and I think it doesnt handle exceptions hierarchy.
In Spring you can declare `noRollbackFor=SomeException.class` and it won't rollback for all subclasses of SomeException.

I'm not sure if we really need this `@AfterThrowing` feature. Maybe just `@AfterReturning` and javers.commit() only when a method is executed without any exception?